components/org.wso2.carbon.identity.auth.otp.core/src/main/java/org/wso2/carbon/identity/auth/otp/core/AbstractOTPAuthenticator.java

```diff
@@ -742,6 +749,42 @@ protected void redirectToOTPLoginPage(AuthenticatedUser authenticatedUser, Strin
         }
     }
 
+    /**
+     * Get remaining number of otp attempts.
+     *
+     * @param authenticatedUser Authenticated User.
+     * @throws AuthenticationFailedException Exception on authentication failure.
+     */
+    private int getRemainingNumberOfOtpAttempts(AuthenticatedUser authenticatedUser, String tenantDomain)
+            throws AuthenticationFailedException {
+
+        try {
+            UserStoreManager userStoreManager = getUserStoreManager(authenticatedUser);
+            String failedAttemptsClaim = getOTPFailedAttemptsClaimUri();
+            if (userStoreManager == null) {
+                throw handleAuthErrorScenario(ERROR_CODE_ERROR_GETTING_USER_STORE_MANAGER);
+            }
+            String fullQualifiedUsername = authenticatedUser.toFullQualifiedUsername();
+            Map<String, String> claimValues = userStoreManager.getUserClaimValues(MultitenantUtils
+                            .getTenantAwareUsername(fullQualifiedUsername),
+                    new String[]{failedAttemptsClaim}, null);
+            String failedSmsOtpAttempts = claimValues.get(failedAttemptsClaim);
```

Comment:
Is this specific to sms otp? If not shall we generalize the variable name?

Reply:
fixed